### PR TITLE
fix(terraform): Adding both azurerm_linux_web_app_slot & azurerm_windows_web_app_slot in scope of the test CKV_AZURE_153

### LINKED ISSUE #5667: CKV_AZURE_153 - Not available for linux_web_app_slot & azurerm_windows_web_app_slot
**Describe the issue**
It seems that the best practice CKV_AZURE_153 is not being checked against the following Terraform resources:

- azurerm_linux_web_app_slot
- azurerm_windows_web_app_slot

CKV_AZURE_153 is used to "Ensure web app redirects all HTTP traffic to HTTPS in Azure App Service"

**Examples**

Same as with "azurerm_app_service_slot"

**Version (please complete the following information):**
N/A

**Additional context**

It seems that the check needs to be adjusted cc https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/app_service_slot

image

### PR DESCRIPTION
**Pull request by tdefise**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

## Description

Fixes #5667 

### Description

Same as for azurerm_app_service_slot

### Fix

Same as for azurerm_app_service_slot

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] I have added tests that prove my feature, policy, or fix is effective and works
- [ ] New and existing tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/terraform/checks/resource/azure/AppServiceEnvironmentZoneRedundant.py
Comment (gruebel) [inline checkov/terraform/checks/resource/azure/AppServiceEnvironmentZoneRedundant.py:1]: probably wrong PR? 😄

File: tests/terraform/checks/resource/azure/test_AppServiceSlotHTTPSOnly.py [hunk 1]
```diff
@@ -20,18 +20,23 @@ def test(self):
...
             "azurerm_app_service_slot.fail2",
+            "azurerm_linux_web_app_slot.fail",
+            "azurerm_windows_web_app_slot.fail",
         }
+        skipped_resources = {}
 
-        passed_check_resources = {c.resource for c in report.passed_checks}
-        failed_check_resources = {c.resource for c in report.failed_checks}
+        passed_check_resources = set([c.resource for c in report.passed_checks])
+        failed_check_resources = set([c.resource for c in report.failed_checks])
```
Comment (gruebel) [inline tests/terraform/checks/resource/azure/test_AppServiceSlotHTTPSOnly.py:35]: please keep the original one.

File: checkov/terraform/checks/resource/azure/AppServiceEnvironmentZoneRedundant.py
Comment (tdefise) [reply checkov/terraform/checks/resource/azure/AppServiceEnvironmentZoneRedundant.py:1]: Yep, I adjusted

File: tests/terraform/checks/resource/azure/test_AppServiceSlotHTTPSOnly.py [hunk 2]
```diff
@@ -20,18 +20,23 @@ def test(self):
 
         passing_resources = {
             "azurerm_app_service_slot.pass",
+            "azurerm_linux_web_app_slot.pass"
+            "azurerm_windows_web_app_slot.pass"
```
Comment (gruebel) [inline tests/terraform/checks/resource/azure/test_AppServiceSlotHTTPSOnly.py:24]:
```suggestion
            "azurerm_linux_web_app_slot.pass",
            "azurerm_windows_web_app_slot.pass",
```

Comment (gruebel) [reply tests/terraform/checks/resource/azure/test_AppServiceSlotHTTPSOnly.py:24]: then the test will pass

Comment (tdefise) [reply tests/terraform/checks/resource/azure/test_AppServiceSlotHTTPSOnly.py]: Fixed

**Comment by tdefise** [comment]
@gruebel could it be because I didn't added check for when "https_only" is not mentionned within azurerm_linux_web_app_slot & azurerm_windows_web_app_slot

By default, it's enabled as it doesn't sais the opposite on Terraform's website https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/linux_web_app_slot#https_only

For  azurerm_app_service_slot it's different https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/app_service_slot#https_only

**Comment by gruebel** [comment]
its ok, let's enforce it 🙂 but you can easily fix the failing tests just adjust the value in `tests/terraform/image_referencer/test_runner_azure_resources.py`, `13` -> `14`

**Comment by tdefise** [comment]
Hi @gruebel,

I would love to not enforce it and learn why changing from 13 to 14 would solve the issue.
I'm not that much familiar with unit testing (just what is required to contribute).

Hope you understand,
Thomas

**Comment by gruebel** [comment]
hey @tdefise good question 🙂 

This specific test has the one of the `slot` resources and therefore each new check or adding the resource type to an existing checks raises the failed or passed check count.

**Comment by tdefise** [comment]
Thanks for the context @gruebel,

I will change that file as well as part of this PR.

Thomas

**Comment by Saarett** [comment]
Hey @tdefise , could you please adjust the file `tests/terraform/image_referencer/test_runner_azure_resources.py `, to fix the failing tests? Thanks 😄

**Comment by tdefise** [comment]
Hey @Saarett,

I actually changed the file in a previous commit, I will have a look at it again.

**Comment by tdefise** [comment]
Should be good now

**Review by gruebel — APPROVED** [review]
thanks, as always 🍸

**Review by rotemavni — APPROVED** [review]

**Pull request merged**